standards/core/rln-contract.md

```diff
@@ -0,0 +1,204 @@
+    3. Extend the membership. The user sends a transaction to re-enable their membership for another expiration term. The user only covers gas costs and does not have to provide additional funds. The original deposit remains in the contract. The membership is prolonged for another expiration term under the same conditions.
+
+The user can extend the membership both during and after its grace period.
+By not extending the membership within its grace period, the user assumes the risk of the membership being taken over at any moment.
+
+A user can hold multiple memberships.
+Memberships are not transferable.
+There is no limitation on the number of membership per user (apart from the global limits, see parameter table).
+
+## Contract governance and mutability
+
+The contract governance is follows:
+
+1. the first deployment of the contract allows the `Owner` to modify certain parameters (TBD) under certain constraints (TBD);
+2. at some point, the `Onwer` would renounce their privileges, and the contract will become immutable;
+3. further upgrades, if necessary, can be done by deploying a new contract and moving the membership set.
+
+## Parameters
+
+We make the following design choices:
+- there are `3` membership tiers: low-, mid-, and high-tier;
+- prices are quoted in `USD`;
+- payment is accepted in `DAI`;
+- the price for a given tier is linearly proportional to its rate limit.
+
+For the initial deployment, we suggest the following values.
+
+| Parameter                                                 | Symbol     | Value   | Units                             | Modifiable? | Comment                        |
+| --------------------------------------------------------- | ---------- | ------- | --------------------------------- | ----------- | ------------------------------ |
+| Epoch length                                              | `epoch`    | `10`    | minutes                           | TBD         |                                |
+| Maximum total rate limit of concurrent active memberships | `R_{max}`  | TBD     | messages per `epoch`              | Yes         | Can be replaced with `N_{max}` |
+| Maximum number of concurrent active memberships           | `N_{max}`  | `10000` | units                             | Yes         | Can be replaced with `R_{max}` |
+| Rate limit for low-tier                                   | `R_{low}`  | `20`    | messages per `epoch`              | TBD         |                                |
+| Rate limit for mid-tier                                   | `R_{mid}`  | `200`   | messages per `epoch`              | TBD         |                                |
+| Rate limit for high-tier                                  | `R_{high}` | `600`   | messages per `epoch`              | TBD         |                                |
+| Unit price                                                | `p_u`      | TBD     | `USD` per `1` message per `epoch` | Yes         |                                |
```

Comment:
What are we thinking?

Unit price of 1c:

- low-tier: 20c
- mid-tier: $2
- high-tier: $6

Per quarter

Reply:
BTW: I'll redefine the unit price to be "per membership expiration term", which is 90 days.